### PR TITLE
Add new registration features: registration on projections and user defined resolution levels. Also channel order fix.

### LINKED ISSUE #7: Fix Channel ordering
Channel indeces sometimes re-ordered. Fix in multiview stitcher.

### PR DESCRIPTION
**Pull request by m-albert**
This PR adds the registration features described in
- #6 
- #5

It also fixes #7 by ensuring the right channel order when retrieving the dask array from the `xarray` based output of `multiview-stitcher`.

@jluethi @nrepina feel free to suggest other parameter names of the new features in case you find these inconvenient!

### CODE REVIEW

File: src/fractal_ome_zarr_hcs_stitching/stitching_task.py [hunk 1]
```diff
@@ -85,78 +85,114 @@ def stitching_task(
...
+    input_spatial_dims = si_utils.get_spatial_dims_from_sim(
+        xim_well_reg.squeeze(drop=True)
+    )
+
+    # determine whether to perform registration on maximum projection in Z
+    reg_max_project_z = registration_on_z_proj and ("z" in input_spatial_dims)
+
+    if reg_max_project_z:
+        xim_well_reg = xim_well_reg.max("z")
+    else:
```
Comment (jluethi) [inline src/fractal_ome_zarr_hcs_stitching/stitching_task.py:109]: Pure style comment, but you can drop this else, it has no effect :)

File: src/fractal_ome_zarr_hcs_stitching/stitching_task.py [hunk 2]
```diff
@@ -85,78 +85,114 @@ def stitching_task(
...
+                ndim=3, t_coords=affine.coords["t"] if "t" in affine.dims else None
+            )
+            affine_3d.loc[{pdim: affine.coords[pdim] for pdim in affine.dims}] = affine
+            affine = affine_3d
+
+        msi_utils.set_affine_transform(
+            msims_fusion[itile], affine, fusion_transform_key
+        )
+
     # FIXME: Something in the fusion changes channel index order. Unclear what
```
Comment (jluethi) [inline src/fractal_ome_zarr_hcs_stitching/stitching_task.py:190]: If the channel order is now preserved, we can drop this FIXME 🎉

**Review by jluethi — APPROVED** [review]
@m-albert This looks awesome! I had 2 minor comments that are more style than anything else. Great to have the channel order maintained and I like the variable names! And super neat how simple the MIP-based registration appears within this task now

I suggest we go ahead with merging this to make it easier to then add new testing functionality on top of that based on Nicole's Zenodo datasets! :)

File: src/fractal_ome_zarr_hcs_stitching/stitching_task.py (hunk 1 above)
Comment (jluethi) [reply src/fractal_ome_zarr_hcs_stitching/stitching_task.py:109]: (i.e. drop the else and the line after it)

Comment (m-albert) [reply src/fractal_ome_zarr_hcs_stitching/stitching_task.py:109]: Oh right, well spotted 🙏 This is a leftover from some renaming 😁

**Comment by m-albert** [comment]
Thanks for your review and comment @jluethi!

Great, let's get this in to have it available for testing with the datasets uploaded by Nicole.

Merging now.

**Pull request merged**